addons/mail/static/tests/discuss_app/discuss.test.js

```diff
@@ -1423,6 +1423,77 @@ test("inbox notifs shouldn't play sound nor open chat bubble", async () => {
     await waitForSteps([]); // no sound alert whatsoever
 });
 
+test("receive new chat from chat plays sound only when out-of-focus", async () => {
+    const pyEnv = await startServer();
+    const partnerId = pyEnv["res.partner"].create({ name: "Dumbledore" });
+    const userId = pyEnv["res.users"].create({ partner_id: partnerId });
+    const channelId = pyEnv["discuss.channel"].create({
+        channel_member_ids: [
+            Command.create({ partner_id: serverState.partnerId }),
+            Command.create({ partner_id: partnerId }),
+        ],
+        channel_type: "chat",
+    });
+    let isFocused = false;
+    mockService("presence", { isOdooFocused: () => isFocused });
+    mockService("mail.sound_effects", {
+        play(soundEffectName, ...args) {
+            asyncStep(`sound:${soundEffectName}`);
+            return super.play(soundEffectName, ...args);
+        },
+    });
+    onRpcBefore("/mail/data", async (args) => {
+        if (args.fetch_params.includes("init_messaging")) {
+            asyncStep("init_messaging");
+        }
+    });
+    await start();
+    await contains(".o_menu_systray i[aria-label='Messages']");
+    await waitForSteps(["init_messaging"]);
+    // simulate receiving a new message with odoo out-of-focused
+    await withUser(userId, () =>
+        rpc("/mail/message/post", {
+            post_data: {
+                body: "New message",
+                message_type: "comment",
+            },
+            thread_id: channelId,
+            thread_model: "discuss.channel",
+        })
+    );
+    await contains(".o-mail-ChatBubble .badge:contains(1)");
+    await waitForSteps(["sound:new-message"]);
+    // simulate odoo tab has focus + receiving a new message
+    isFocused = true;
+    await withUser(userId, () =>
+        rpc("/mail/message/post", {
+            post_data: {
+                body: "New message",
+                message_type: "comment",
+            },
+            thread_id: channelId,
+            thread_model: "discuss.channel",
+        })
+    );
+    await contains(".o-mail-ChatBubble .badge:contains(2)");
+    await animationFrame();
+    await waitForSteps([]);
+    // simulate odoo tab has no focus + receiving a new message
+    isFocused = false;
+    await withUser(userId, () =>
+        rpc("/mail/message/post", {
+            post_data: {
+                body: "New message",
+                message_type: "comment",
+            },
+            thread_id: channelId,
+            thread_model: "discuss.channel",
+        })
+    );
+    await contains(".o-mail-ChatBubble .badge", { text: "3" });
+    await waitForSteps(["sound:new-message"]);
+});
```

Comment:
Last `contains()` takes 3 seconds (timeout to find in DOM)... That's unfortunate :(

Reply:
In the end I removed assertion of chat bubble. Still mutation is not observed that needs fixing at some point.